### PR TITLE
Feature/otter theme entrypoint

### PR DESCRIPTION
**Pull request by kpanot**
## Proposed change

Remove Otter Theming and expose it as sub entries to remove the dependency to Angular Material

## Related issues

- :rocket: PR #889

### CODE REVIEW

File: packages/@o3r/styling/schematics/ng-update/v10.0/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      initialTree.create('/style.scss', content);
+      const runner = new SchematicTestRunner('migrations', collectionPath);
+      const tree = await lastValueFrom(runner.callRule(updateScssImports(), initialTree));
+      const res = tree.readText('/style.scss').replaceAll(/^ +$/gm, '').replaceAll(/\n\n/g, '\n');
+
+      expect(res).toEqual(expected);
+    });
+
+    it('should update a file with Otter theme usage and clean previous dep', () => {
+
```
Comment (fpaul-1A) [inline packages/@o3r/styling/schematics/ng-update/v10.0/index.spec.ts:76]: missing test ?

**Review by fpaul-1A — DISMISSED** [review]

File: packages/@o3r/styling/schematics/ng-update/v10.0/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      expect(res).toEqual(expected);
+    });
+
+    it('should update a file with Otter theme usage with non-standard alias', async () => {
+      const content = `@use "@o3r/styling" as test-name;
+        $var: test-name.meta-theme-to-otter();
+        $var2: test-name.var(#000);`;
+      const expected = `@use '@o3r/styling/otter-theme' as otter-theme;
+        $var: otter-theme.meta-theme-to-otter();`;
```
Comment (cpaulve-1A) [inline packages/@o3r/styling/schematics/ng-update/v10.0/index.spec.ts:67]: You forgot to handle your var2

File: docs/styling/THEME.md [hunk 3]
```diff
@@ -256,7 +256,7 @@ as a parameter:
 ```scss
 // _styling.scss
 @use '@angular/material' as mat;
-@use '@o3r/styling' as o3r;
+@use '@o3r/styling/otter-theme' as otter-theme;
```
Comment (matthieu-crouzet) [inline docs/styling/THEME.md:259]:
```suggestion
@use '@o3r/styling/otter-theme';
```
We don't need the `as` if we don't rename it :)

File: packages/@o3r/styling/schematics/ng-update/v10.0/update-scss-imports.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        };
+      })
+      .filter(({ toReplace, importName }) => !!importName && !!toReplace?.length)
+      .forEach(({ file, content, toReplace, importName }) => {
+        let newContent = `@use '@o3r/styling/otter-theme' as ${otterSubEntryPrefix};\n` +
+          toReplace!.reduce((acc, {from, to}) => {
+            return acc.replaceAll(from, to);
+          }, content);
+        if (newContent.indexOf(`${importName!}.`) === -1) {
+          newContent = newContent.replace(new RegExp(`@use +['"]@o3r/styling['"] +as +${importName!};\n?`), '');
```
Comment (cpaulve-1A) [inline packages/@o3r/styling/schematics/ng-update/v10.0/update-scss-imports.ts:54]: won't you remove also the import for legit methods?

File: packages/@o3r/styling/schematics/ng-update/index.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable @typescript-eslint/naming-convention */
```
Comment (matthieu-crouzet) [inline packages/@o3r/styling/schematics/ng-update/index.ts:1]: why disabling these rules ?

File: docs/styling/THEME.md (hunk 3 above)
Comment (kpanot) [reply docs/styling/THEME.md:259]: That's true but I keep it as good practice

File: packages/@o3r/styling/schematics/ng-update/v10.0/update-scss-imports.ts (hunk 4 above)
Comment (kpanot) [reply packages/@o3r/styling/schematics/ng-update/v10.0/update-scss-imports.ts:54]: nop, thanks to the `if` just above.
We remove the import only if it is in the new content

File: docs/styling/THEME.md [hunk 6]
```diff
@@ -357,26 +357,33 @@ This will allow a component level customization.
...
+The principal functions are the following:
+
+* **otter-theme.color(*`<theme-palette>`*, *`<value>`*)**: similar to [mat-color](https://material.angular.io/guide/theming-your-components), it will retrieve the color from the theme palette. Instead of printing directly the color, the function will generate a css-var (ex: `otter-theme.color($primary-palette, 500)` will generate `var('--primary-color-500', #050)`).
+* **otter-theme.contrast(*`<theme-palette>`*, *`<value>`*)**: similar to [mat-contrast](https://material.angular.io/guide/theming-your-components), it will retrieve the color from the theme palette. Instead of printing directly the color, the function will generate a css-var (ex: `otter-theme.contrast($primary-palette, 500)` will generate `var('--primary-color-contrast-500', #505)`).
 
 ```scss
 //file: ./my-component-pres.style.theme.scss
 
 @use '@o3r/styling' as o3r;
+@use '@o3r/styling/otter-theme' as otter-theme;
```
[the remark this answers is not in the document]
Comment (kpanot) [reply docs/styling/THEME.md:380]: I keep it as good practice

File: packages/@o3r/styling/schematics/ng-update/index.ts (hunk 5 above)
Comment (kpanot) [reply packages/@o3r/styling/schematics/ng-update/index.ts:1]: Wrong copy past, I remove it

File: packages/@o3r/styling/schematics/ng-update/v10.0/index.spec.ts (hunk 2 above)
Comment (kpanot) [reply packages/@o3r/styling/schematics/ng-update/v10.0/index.spec.ts:67]: pushed to quick before leaving the office, I fixed it

**Review by matthieu-crouzet — APPROVED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Review by cpaulve-1A — APPROVED** [review]

**Pull request merged**